aws-opentelemetry-distro/src/amazon/opentelemetry/distro/exporter/otlp/aws/metrics/otlp_aws_emf_exporter.py

```diff
@@ -0,0 +1,482 @@
+                emf_log[f"resource.{key}"] = str(value)
+
+        # Initialize collections for dimensions and metrics
+        all_attributes = {}
+        metric_definitions = []
+
+        # Process each metric record
+        for record in metric_records:
+            # Collect attributes from all records (they should be the same for all records in the group)
+            if hasattr(record, "attributes") and record.attributes:
+                for key, value in record.attributes.items():
+                    all_attributes[key] = value
+
+            metric_name = self._get_metric_name(record)
+
+            # Skip processing if metric name is None or empty
+            if not metric_name:
+                continue
+
+            unit = self._get_unit(record.instrument)
+
+            # Create metric data dict
+            metric_data = {}
+            if unit:
+                metric_data["Unit"] = unit
+
+            # Process gauge metrics (only type supported in PR 1)
+            if hasattr(record, "value"):
+                # Store value directly in emf_log
+                emf_log[metric_name] = record.value
+
+            # Add to metric definitions list
+            metric_definitions.append({"Name": metric_name, **metric_data})
+
+        # Get dimension names from collected attributes
+        dimension_names = self._get_dimension_names(all_attributes)
+
+        # Add attribute values to the root of the EMF log
+        for name, value in all_attributes.items():
+            emf_log[name] = str(value)
+
+        # Add the single dimension set to CloudWatch Metrics if we have dimensions and metrics
+        if dimension_names and metric_definitions:
+            emf_log["_aws"]["CloudWatchMetrics"].append(
+                {"Namespace": self.namespace, "Dimensions": [dimension_names], "Metrics": metric_definitions}
+            )
+
+        return emf_log
+
+    # pylint: disable=no-member
+    def _send_log_event(self, log_event: Dict):
+        """
+        Send a log event to CloudWatch Logs.
+
+        Basic implementation for PR 1 - sends individual events directly.
+        """
+        try:
+            # Create log group and stream if they don't exist
+            try:
+                self.logs_client.create_log_group(logGroupName=self.log_group_name)
+                logger.debug("Created log group: %s", self.log_group_name)
+            except ClientError as error:
+                # Check if it's a ResourceAlreadyExistsException (botocore exception handling)
+                if error.response.get("Error", {}).get("Code") == "ResourceAlreadyExistsException":
+                    logger.debug("Log group %s already exists", self.log_group_name)
+
+            # Create log stream if it doesn't exist
+            try:
+                self.logs_client.create_log_stream(logGroupName=self.log_group_name, logStreamName=self.log_stream_name)
+                logger.debug("Created log stream: %s", self.log_stream_name)
```

Comment:
Similarly, we should also ensure the log stream exists from the `_ensure_log_group_exists` method, and don't need to attempt to create a log stream every time we send a log batch.

Reply:
Is this required to be here because it is possible that the user can delete their log stream?